hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/SparkReaderContextFactory.java

```diff
@@ -107,7 +108,8 @@ public HoodieReaderContext<InternalRow> getContext() {
           sparkParquetReader,
           JavaConverters.asScalaBufferConverter(filters).asScala().toSeq(),
           JavaConverters.asScalaBufferConverter(filters).asScala().toSeq(),
-          new HadoopStorageConfiguration(configurationBroadcast.getValue().value()));
+          new HadoopStorageConfiguration(configurationBroadcast.getValue().value()),
+          tableConfig);
```

Comment:
`tableConfig` can be fetched in the constructor by `metaClient.getTableConfig()` for consistency and the method argument can be removed.

Reply:
updated